### PR TITLE
CRUD: paging on empty pages

### PR DESCRIPTION
**Pull request by BearLemma**
This PR does two things.
1) it provides previous/next paging for all pages (even the first one). The rationale is twofold for this. a) it's simpler and b) there actually always can be some results anywhere as new instances can be inserted anytime while paging.
2) It uses the concept of cursor reversal to be able to come back from empty pages (during reversal of a cursor, the direction is reversed and it's made inclusive so that it includes the pivot (last/first element) of the page we came from). 

TODO:

- [x] Add a bit more test coverage for the new behavior.

### CODE REVIEW

File: server/src/datastore/crud.rs [hunk 1]
```diff
@@ -87,12 +87,7 @@ fn get_next_page(
     host: &Option<String>,
     results: &[JsonObject],
 ) -> Result<Option<Url>> {
-    if results.is_empty() {
-        return Ok(None);
-    }
-
-    let has_bwd_cursor = query.cursor.as_ref().map(|c| !c.forward).unwrap_or(false);
-    if query.page_size == results.len() as u64 || has_bwd_cursor {
+    if !results.is_empty() {
```
Comment (espindola) [inline server/src/datastore/crud.rs]: The commit message says "Always provide next/prev page", but the patch doesn't do that for empty pages, which one is wrong?

File: server/src/datastore/crud.rs [hunk 2]
```diff
@@ -109,12 +104,7 @@ fn get_prev_page(
     host: &Option<String>,
     results: &[JsonObject],
 ) -> Result<Option<Url>> {
-    if results.is_empty() || query.cursor.is_none() {
-        return Ok(None);
-    }
-
-    let has_fwd_cursor = query.cursor.as_ref().map(|c| c.forward).unwrap_or(false);
-    if query.page_size == results.len() as u64 || has_fwd_cursor {
+    if !results.is_empty() {
```
Comment (espindola) [inline server/src/datastore/crud.rs:112]: Same here.

File: server/src/datastore/crud.rs (hunk 1 above)
Comment (espindola) [reply server/src/datastore/crud.rs]: Can you expand the commit message a bit on the why of the patch?

Comment (BearLemma) [reply server/src/datastore/crud.rs]: Added more info to the commit message.

**Review by espindola — APPROVED** [review]

**Pull request merged**